### PR TITLE
[CEDS-3024] Adding two way messaging service inbox

### PR DESCRIPTION
**Pull request by glidester**
Adding the new message inbox to the service that is
linked to from the new 'what_do_you_want_to_do' choice
page.

### CODE REVIEW

File: app/connectors/SecureMessageFrontendConnector.scala [hunk 1]
```diff
@@ -0,0 +1,118 @@
...
+          logger.warn(
+            s"Received a ${exc.statusCode} response from secure-messaging-frontend service while retrieving the user's inbox. ${exc.message}"
+          )
+
+          Future.failed(exc)
+      }*/
+}
+
+object SecureMessageFrontendConnector {
+  def fakeInboxResponse(): Future[InboxPartial] = Future.successful(InboxPartial(inboxPartial))
```
Comment (udinobi) [inline app/connectors/SecureMessageFrontendConnector.scala:54]: `fakeInboxResponse` could also be a `lazy val`.

File: test/controllers/SecureMessagingControllerSpec.scala [hunk 2]
```diff
@@ -0,0 +1,101 @@
...
+  override def beforeEach(): Unit = {
+    super.beforeEach()
+
+    reset(partialWrapperPage)
+    secureMessagingFeatureAction.reset()
+    when(partialWrapperPage.apply(any[HtmlFormat.Appendable])(any[Request[_]], any[Messages])).thenReturn(HtmlFormat.empty)
+  }
+
+  override def afterEach(): Unit =
+    super.afterEach()
```
Comment (udinobi) [inline test/controllers/SecureMessagingControllerSpec.scala:58]: :thinking: Do you really need to override `afterEach` here?

File: app/connectors/SecureMessageFrontendConnector.scala (hunk 1 above)
Comment (glidester) [reply app/connectors/SecureMessageFrontendConnector.scala]: Changed :+1:

File: test/controllers/SecureMessagingControllerSpec.scala (hunk 2 above)
Comment (glidester) [reply test/controllers/SecureMessagingControllerSpec.scala]: Removed :+1:

File: app/config/AppConfig.scala [hunk 3]
```diff
@@ -111,3 +112,8 @@ case class TrackingConsentFrontend(gtm: Gtm, url: String)
 case class Platform(frontend: Frontend)
 
 case class Frontend(host: Option[String])
+
+case class SecureMessaging(protocol: Option[String], host: String, port: Option[Int], fetchInbox: String) {
+  def fetchInboxEndpoint(): String =
```
Comment (udinobi) [inline app/config/AppConfig.scala:117]: Sorry, Tim :upside_down_face: ... maybe worth also set this as `lazy val` rather than `def` ?

Comment (glidester) [reply app/config/AppConfig.scala]: Ok, changed

**Review by udinobi — APPROVED** [review]

**Pull request merged**